### PR TITLE
Added greatness, power, and rating to the contract.

### PR DESCRIPTION
**Pull request by treppers**
Also updated documentation as well as some fix minor typos.

### CODE REVIEW

**Review by 5p0rt5BEArD — APPROVED** [review]
These are great additions that I approve of but I've commented with potential further improvements. 

### Access via token ID

_**Access that works with either Item index or tokenID**_

Access via token ID is great and inline with what I was expecting. They weren't needed for The Crypt because the item index was already known. In future I think it would be good for all classifications that can be derived from item index (rank, material, class, level) to have access or taking index AND an accessor taking token ID. The token ID version would always use the component functions to get the item index and then call the associated item index accessor.

_**Access that works with only tokenID**_
Access to Greatness and Rating can only be via a tokenID because that's needed to seed the random function. So no additions needed here.

**Classification Names**
The original set of classification names (material, class, rank) came from an older version of https://docs.google.com/spreadsheets/d/118x6e4WEVZ7q-XvsRoyDNq7Nufw3xSmMKeYxPpivwRA/edit#gid=671005165

For Rank, that spreadsheet now has "Item Power (high is best)" in one sheet and is references as Value in another. 

I'm not sure if there is documented use of the new terms: Level (inverse of Rank) or Rating (Level * Greatness). And I think I've seen reference to Rating as Power elsewhere.

In summary its a bit confused and it would be good to be able to link to a document that uses terms that match this contract for all of

Class
Material
Rank (should this be Power?)
Level (should this be Power?)
Rating

**Comment by treppers** [comment]
### **Clarity on new classifications**
Sorry for the confusion.  There was some back and forth and _Power_ was in the running for what eventually became _Level_. 
 which is why there are remnants of the word _power_ scattered within the comments.  I did another search within the code and cleaned up remnants of the word _power_. The final list of classifications should be:
- **Class** - Derived from component index
- **Material** - Derived from component index
- **Rank** - Derived from component index
- **Level** - 6 - rank (if Neck or Ring, 4 - rank)
- **Greatness** - Derived from pluck function
- **Rating** - Level * Greatness (This should be the ultimate "at-a-glance" metric for the item/bag)

### **Access via tokenID or Item index**
I could easily add the alternative access methods for getClass, getMaterial, getRank, and getLevel.  However do you have any suggestions on what to call the differing functions?

Proposal 1: Append a suffix to both functions so that suffix matches the input method (i.e. getRankByTokenID and getRankByItemIndex)

Proposal 2: Append one of the input methods with a suffix while the other input method gets the "nonsuffix'ed" function name (i.e getRankByItemIndex and getRank)

### Bug in Ranking for Neck and Ring
While unit testing a LootBagClassification contract, I found that Neck and Ring were both being calculated incorrectly.  Neck should all return a ranking of 1, while Ring should return a ranking of 1,2,3,1,1.  See [loot.foundation definition](https://docs.loot.foundation/canonical-principles/loot/item-types/jewelry).

Will do another pull request with the cleansed comments and the fix in ranking.

File: LootClassification.sol [hunk 1]
```diff
@@ -162,12 +162,15 @@ contract LootClassification
...
+        if (index > 2)
+            return 1;
+        else 
+            return index + 1;
     }
     
     function getNeckRank(uint256 index) pure public returns (uint256)
     {
-        return 3 - index;
+        return index * 0 + 1;
```
Comment (5p0rt5BEArD) [inline LootClassification.sol]: why not return 1?

**Review by 5p0rt5BEArD — COMMENTED** [review]
Thanks for the power/level classification

Re the tokenID / itemIndex accessors, I like your proposal 2. I think the default using tokenID is good and using the _ByItemIndex version is the special case and internal version.

**Comment by treppers** [comment]
Oh hah, forgot I left that in there.  My IDE didn't like the fact I was
declaring index and then not using it.  And then I didn't want to change
the declaration for just this one function to not require index.

I'll just change it to return 1 and then tell my IDE to ignore it.
